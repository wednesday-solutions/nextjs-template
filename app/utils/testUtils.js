import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from '../configureStore';
import colors from '@themes/colors';
import { translationMessages, DEFAULT_LOCALE } from '../i18n';

export const getComponentStyles = (Component, props = {}) => {
  renderProvider(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);

  return window.getComputedStyle(componentRoots[0])._values;
};

export const renderProvider = (children) => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={colors}>{children}</ThemeProvider>
    </IntlProvider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const renderStoreProvider = (children) => {
  const store = configureStore({});
  return render(
    <Provider store={store}>
      <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
        <ThemeProvider theme={colors}>{children}</ThemeProvider>
      </IntlProvider>
    </Provider>
  );
};
