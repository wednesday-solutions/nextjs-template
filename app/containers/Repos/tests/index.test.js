/**
 *
 * Tests for Repos container
 *
 *
 */

import React from 'react'
// import { fireEvent } from '@testing-library/dom';
import { renderProvider } from '@utils/testUtils'
import { ReposTest as Repos } from '../index'

describe('<Repos /> container tests', () => {
  // let submitSpy

  beforeEach(() => {
    // submitSpy = jest.fn();
  })

  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<Repos recommendations={[]} />)
    expect(baseElement).toMatchSnapshot()
  })
})
