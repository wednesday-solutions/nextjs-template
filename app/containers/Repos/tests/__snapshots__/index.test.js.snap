// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repos /> container tests should render and match the snapshot 1`] = `
.emotion-0.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 500px;
  width: 100%;
  margin: 0 auto;
  padding: 20px;
}

.emotion-2 {
  white-space: pre-line;
  font-size: 1.75rem;
  font-weight: 500;
}

.emotion-5 {
  text-align: right;
}

.emotion-5.emotion-5 span {
  color: #006ED6;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-5.emotion-5 span:hover {
  opacity: 0.8;
}

.emotion-6 {
  white-space: pre-line;
}

.emotion-8.emotion-8 {
  margin: 20px 0;
  max-width: 500;
}

.emotion-15.emotion-15 {
  margin: 20px 0;
  color: grey;
  color: grey;
}

<body>
  <div>
    <div
      class="emotion-0"
      style="height: 100vh; align-self: center;"
    >
      <div
        class="ant-row emotion-1"
      >
        <span
          class="emotion-2"
          data-testid="text"
        >
          Recommended
        </span>
      </div>
      <div
        class="ant-row ant-row-space-between emotion-1"
      >
        <div
          class="ant-row emotion-1"
          data-testid="recommended"
        />
        <a
          class="emotion-5"
          href="https://www.iamawesome.com/"
        >
          <span
            class="emotion-6"
            data-testid="text"
          >
            You are awesome
          </span>
        </a>
      </div>
      <div
        class="ant-divider emotion-1 ant-divider-horizontal"
        role="separator"
      />
      <div
        class="ant-card ant-card-bordered emotion-8 emotion-1"
        maxwidth="500"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              Repository Search
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <span
            class="emotion-6"
            data-testid="text"
          >
            Get details of repositories
          </span>
          <span
            class="ant-input-group-wrapper ant-input-group-wrapper-outlined emotion-1 ant-input-search"
          >
            <span
              class="ant-input-wrapper ant-input-group emotion-1"
            >
              <input
                class="ant-input emotion-1 ant-input-outlined"
                data-testid="search-bar"
                type="text"
                value=""
              />
              <span
                class="ant-input-group-addon"
              >
                <button
                  class="ant-btn emotion-1 ant-btn-default ant-btn-icon-only ant-input-search-button"
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <span
                      aria-label="search"
                      class="anticon anticon-search"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="search"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        class="ant-card ant-card-bordered emotion-15 emotion-1"
        color="grey"
        data-testid="error-state"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              Repository List
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <span
            class="emotion-6"
            data-testid="text"
          >
            Search for a repository by entering it's name in the search box
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
